llvm/lib/Target/SystemZ/SystemZAsmPrinter.cpp

```diff
@@ -1112,6 +1113,23 @@ void SystemZAsmPrinter::emitEndOfAsmFile(Module &M) {
   if (TT.isOSzOS()) {
     emitADASection();
     emitIDRLSection(M);
+    // Emit EXTRN declarations.
+    OutStreamer->pushSection();
+    for (auto &GO : M.global_objects()) {
```

Comment:
It's a bit unfortunate if we have to have this loop here.  It looks like `AsmPrinter::doFinalization` is already doing very similar things for other platforms, might be nicer if we could hook into that.